backend/tests/controllers/reports/test_reports_test_executions.py

```diff
@@ -0,0 +1,197 @@
+    test_execution = generator.gen_test_execution(
+        artefact_build, environment, created_at=datetime.now()
+    )
+    generator.gen_artefact_build_environment_review(artefact_build, environment)
+    generator.gen_test_event(test_execution, "provision_fail")
+
+    response = test_client.get(
+        "/v1/reports/test-executions",
+        params={
+            "start_date": (datetime.now() - timedelta(days=1)).isoformat(),
+            "include_test_events": True,
+        },
+    )
+
+    table = _read_csv_response(response)
+
+    assert len(table) == 2
+    assert table[0] == [
+        str(column)
+        for column in [
+            *TEST_EXECUTIONS_REPORT_COLUMNS,
+            "TestEvent.event_name",
+            "TestEvent.timestamp",
+            "TestEvent.detail",
+        ]
+    ]
+    assert table[1] == _expected_report_row(
+        test_execution, db_session=db_session, include_test_events=True
+    )
+
+
+def _read_csv_response(response: Response) -> list:
+    content = response.content.decode()
+    csv_reader = csv.reader(StringIO(content))
+    return list(csv_reader)
+
+
+def _expected_report_row(
+    test_execution: TestExecution,
+    db_session: Session,
+    include_test_events: bool = False,
+) -> list:
+    environment = test_execution.environment
+    artefact = test_execution.artefact_build.artefact
+    family = artefact.stage.family
+    environment_review = (
+        db_session.query(ArtefactBuildEnvironmentReview)
+        .filter(
+            ArtefactBuildEnvironmentReview.artefact_build_id
+            == test_execution.artefact_build_id,
+            ArtefactBuildEnvironmentReview.environment_id
+            == test_execution.environment_id,
+        )
+        .one()
+    )
+
+    expected_report = [
+        family.name,
+        str(artefact.id),
+        artefact.name,
+        artefact.version,
+        artefact.status.name,
+        artefact.track,
+        artefact.series,
+        artefact.repo,
+        str(artefact.created_at),
+        str(test_execution.id),
+        test_execution.status.name,
+        "" if not test_execution.ci_link else test_execution.ci_link,
+        "" if not test_execution.c3_link else test_execution.c3_link,
+        "" if not test_execution.checkbox_version else test_execution.checkbox_version,
+        environment.name,
+        environment.architecture,
+        str(environment_review.review_decision),
+        environment_review.review_comment,
+    ]
+
+    if include_test_events:
+        test_events = test_execution.test_events
+        expected_report.extend(
+            [
+                str([te.event_name for te in test_events]),
+                str(
+                    [
+                        te.timestamp.strftime("%Y-%m-%d %H:%M:%S:%f")[:-3]
+                        for te in test_events
+                    ]
+                ),
+                str([te.detail for te in test_events]),
+            ]
+        )
```

Comment:
I look at this, and I thought a better alternative would be to provide a list of dictionaries instead, so that the events even without the order in the tuple. What do you think about the use of dict over tuple?

Reply:
dict sounds good to me